### DISCUSSION TITLE
images resolution 2376 × 1584

### DISCUSSION THREAD

**Discussion by ovidedecroly** [ultralytics/ultralytics#14335] [General]
Hi @glenn-jocher 

I hope you are doing well. I have a question regarding the usage of high-resolution images with YOLOv8. Specifically, I am planning to use images with a resolution of 2376×1584 for training, validation, and testing purposes. I am using an A100 GPU for this task. I would like to know if YOLOv8 can effectively handle this resolution for these tasks.

Could you please provide insights or recommendations on the following points:

Are there any limitations or considerations I should be aware of when using YOLOv8 with such high-resolution images?
What impact might this resolution have on the training time and GPU memory usage?
Are there any specific settings or configurations in YOLOv8 that I should adjust to optimize performance with high-resolution images?
Would you recommend any particular strategies to manage potential challenges associated with high-resolution image processing in YOLOv8?
Thank you for your assistance.

Best regards,

**Comment by glenn-jocher** [comment]
Hi @ovidedecroly,

Thank you for reaching out! Using high-resolution images like 2376×1584 with YOLOv8 on an A100 GPU is definitely feasible, but there are a few considerations to keep in mind to optimize performance and manage resources effectively.

### Key Considerations and Recommendations:

1. **GPU Memory Usage**:
   - High-resolution images will significantly increase GPU memory usage. The A100 GPU is quite powerful, but it's still important to monitor memory usage to avoid out-of-memory errors.
   - You can use mixed precision training (`amp=True`) to reduce memory consumption and potentially speed up training.

2. **Training Time**:
   - Higher resolution images will increase the training time per epoch due to the larger amount of data being processed. Consider using a smaller batch size to fit the images into memory, which might help in managing the training time.

3. **Image Size Configuration**:
   - Ensure that the `imgsz` parameter in your training configuration matches your image resolution. For example:
     ```python
     results = model.train(data='your_dataset.yaml', epochs=100, imgsz=(2376, 1584))
     ```

4. **Data Augmentation**:
   - Utilize data augmentation techniques to enhance the robustness of your model. This can help in making the model more generalizable despite the high resolution of the images.

5. **Performance Optimization**:
   - Consider using the `cache` parameter to cache images in memory or on disk, which can speed up the training process:
     ```python
     results = model.train(data='your_dataset.yaml', epochs=100, imgsz=(2376, 1584), cache='ram')
     ```

6. **Monitoring and Adjustments**:
   - Regularly monitor GPU utilization and training metrics. Tools like TensorBoard can be very helpful for this purpose.
   - If you encounter performance bottlenecks, you might need to experiment with different batch sizes and learning rates.

### Example Configuration:
Here's a sample configuration snippet for training with high-resolution images:
```python
from ultralytics import YOLO

# Load a model
model = YOLO('yolov8n.pt')  # or your custom model

# Train the model
results = model.train(
    data='your_dataset.yaml',
    epochs=100,
    imgsz=(2376, 1584),
    batch=4,  # Adjust based on your GPU memory
    amp=True,  # Enable mixed precision training
    cache='ram'  # Cache images in RAM
)
```

### Additional Resources:
For more detailed guidance, you can refer to our [documentation on training](https://docs.ultralytics.com/modes/train) and [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example) to ensure you have all the necessary steps covered.

Feel free to reach out if you have any more questions. Happy training! 🚀

**Comment by ovidedecroly** [comment]
Hi @glenn-jocher 
Glen,

I have two primary sources of data: a public dataset and my own private dataset. I am reaching out to seek your expertise on the best practices for combining these datasets effectively for training the YOLOv8 model.

Here are some details about my datasets:

Public Dataset:
Resolution: Most images are 130x130 pixels.
Quantity per Species: Varies significantly; some species have as few as 5 images, while others have up to 169 images.
Private Dataset:
Resolution: High resolution at 2376x1584 pixels.
Quantity per Species: Ranges from 10 to 54 images per species.

Number of image public dataset per species:
5
75
73
7
79
94
1
32
2
16
48
2
8
43
91
169
3
96
15
141
10
1
10
7
13
89
5
89
61
1
2
3
6
1
61
15
102

number of image private (own) dataset:
12
6
7
7
18
9
54
8
9
9
7
6
9
13
9
7
11
17
9
8
9
12
17
13
10
7
8
11
5
6
10
9
12
9
6
13
11

Given these variations, I am considering several techniques but would appreciate your insights on the following:

Transfer Learning with Fine-Tuning: Using the public dataset for initial training and then fine-tuning with the private dataset. Does this approach make sense given the difference in resolution and quantity of images?

Domain Adaptation: If the distribution between the public and private datasets differs significantly, would you recommend domain adaptation techniques? If so, which methods would be most effective?

Data Splitting Strategy: How should I approach splitting the combined dataset for training and validation? Should I augment all images, or focus augmentation on the underrepresented classes?

Weighted Loss Function: Given the imbalance in the number of images per species, would applying a weighted loss function be beneficial to ensure the model does not favor the more represented classes?

Any Other Recommendations: Are there any other techniques or best practices you would suggest for effectively combining and training on these datasets?

I would greatly appreciate any guidance or suggestions you can provide. Thank you for your time and support.

Best regard

**Reply by glenn-jocher** [reply]
Hi @ovidedecroly,

Thank you for reaching out. Here are some insights and recommendations for effectively combining your public and private datasets for training YOLOv8:

1. **Transfer Learning with Fine-Tuning**: This approach is sensible. Start by training on the public dataset and then fine-tune with your private dataset. YOLOv8 can handle varying resolutions, but you might want to resize images to a consistent size using the `imgsz` parameter during training.

2. **Domain Adaptation**: If the datasets differ significantly, domain adaptation techniques like adversarial training or feature alignment can be beneficial. However, starting with transfer learning and observing the initial results might be a simpler first step.

3. **Data Splitting Strategy**: Ensure a balanced split of training and validation sets. Augmenting underrepresented classes can help balance the dataset. Tools like Albumentations can be useful for this.

4. **Weighted Loss Function**: Applying a weighted loss function can indeed help address class imbalance. YOLOv8 supports class weights, which you can specify in your dataset configuration file.

5. **Other Recommendations**: 
   - **Normalization**: Ensure consistent normalization across datasets.
   - **Augmentation**: Apply augmentations like random crops, flips, and rotations to enhance model robustness.
   - **Batch Size and Learning Rate**: Monitor GPU memory usage and adjust batch size and learning rate accordingly.

For more detailed guidance, you can refer to our [data preprocessing guide](https://docs.ultralytics.com/guides/preprocessing_annotated_data/).

**Reply by ovidedecroly** [reply]
Hi @glenn-jocher,
I hope this message finds you well.

I have a few questions regarding the division of training, validation, and testing data.

When you mention splitting data for training, validation, and testing, do you mean that the validation data is also used for testing purposes simultaneously?

Is data augmentation carried out only on the training data, or is it applied to the validation and testing data as well?

For example, if I have the following data:

Class A: Public data 30, my data 25
Class B: Public data 15, my data 20
In this case, does a balanced division mean a 50:50 split for both the training and validation sets (including the test set)?

Thank you for your assistance.

Best regards,

**Reply by glenn-jocher** [reply]
Hi @ovidedecroly,

I hope you're doing well too!

When we talk about splitting data for training, validation, and testing, the validation data is used during training to tune the model, while the test data is reserved for final evaluation to ensure unbiased performance metrics. They are not used simultaneously for both purposes.

Data augmentation is typically applied only to the training data to help the model generalize better. Applying augmentation to validation or test data would not provide a true measure of the model's performance on real-world data.

Regarding your example, a balanced division doesn't necessarily mean a 50:50 split. Instead, it means ensuring that each class is proportionately represented in both the training and validation sets. For instance, if you have 55 images of Class A and 35 of Class B, you might split them into 70% training and 30% validation, maintaining the class distribution within each split.

I hope this clarifies things! Let me know if you have any more questions.
